ai21/errors.py

```diff
@@ -111,3 +111,10 @@ def __init__(self, chunk: str, error_message: Optional[str] = None):
 class InternalDependencyException(AI21APIError):
     def __init__(self, details: Optional[str] = None):
         super().__init__(530, details)
+
+
+class CodeParsingError(AI21Error):
+    def __init__(self, details: Optional[str] = None):
+        message = f"Code can't be parsed: {details}"
```

Comment:
If you pass `None` here the message would be 
`Code can't be parsed: None` 
That's a strange behavior for the user. 

Perhaps something like this - 
```python
class CodeParsingError(AI21Error):
    """Exception raised when code cannot be parsed successfully."""
    
    def __init__(self, details: Optional[str] = None, message: Optional[str] = None):
        self.message = message or f"Failed to parse code: {details or 'No details provided'}"
        super().__init__(self.message)
```